### ISSUE TITLE
"Code that will be rejected by a future version of Rust"

### ISSUE THREAD

**Issue by vtvz** [influxdb-rs/influxdb-rust#179]
-   Rust Version: nightly-2025-11-02-x86_64-unknown-linux-gnu
-   Runtime version (e.g. Tokio): v1.48.0
-   OS Version: Manjaro Linux 25.0.10

Steps to Reproduce:

1. Run `cargo clippy`
2. Get this error

```
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.28s
warning: the following packages contain code that will be rejected by a future version of Rust: influxdb v0.7.2
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 2`
```

Running `cargo report future-incompatibilities --id 2` shows this report:

<details>
  <summary>Report</summary>

```
The following warnings were discovered during the build. These warnings are an
indication that the packages contain code that will become an error in a
future release of Rust. These warnings typically cover changes to close
soundness problems, unintended or undocumented behavior, or critical problems
that cannot be fixed in a backwards-compatible fashion, and are not expected
to be in wide use.

Each warning should contain a link for more information on what the warning
means and how to resolve it.

to solve this problem, you can try the following approaches:

- ensure the maintainers know of this problem (e.g. creating a bug report if needed)
or even helping with a fix (e.g. by creating a pull request)
  - influxdb@0.7.2
  - repository: https://github.com/influxdb-rs/influxdb-rust
  - detailed warning command: `cargo report future-incompatibilities --id 2 --package influxdb@0.7.2`

- use your own version of the dependency with the `[patch]` section in `Cargo.toml`
For more information, see:
https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section

The package `influxdb v0.7.2` currently triggers the following future incompatibility lints:
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:23:10
>    |
> 23 | #![doc = cargo_toml!(indent="", "derive")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
> 
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:104:9
>     |
>  23 | #![doc = cargo_toml!(indent="", "derive")]
>     |          -------------------------------- in this macro invocation
> ...
> 104 |         cargo_toml_private!($indent, "", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:80:10
>    |
> 80 | #![doc = cargo_toml!(indent="\t", "derive")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
> 
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:104:9
>     |
>  80 | #![doc = cargo_toml!(indent="\t", "derive")]
>     |          ---------------------------------- in this macro invocation
> ...
> 104 |         cargo_toml_private!($indent, "", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:82:10
>    |
> 82 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "reqwest-client")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
> 
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:108:9
>     |
>  82 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "reqwest-client")]
>     |          ------------------------------------------------------------------------------------------- in this macro invocation
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:84:10
>    |
> 84 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "hyper-client")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
> 
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:108:9
>     |
>  84 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "hyper-client")]
>     |          ----------------------------------------------------------------------------------------- in this macro invocation
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:86:10
>    |
> 86 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "curl-client")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
> 
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:108:9
>     |
>  86 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "curl-client")]
>     |          ---------------------------------------------------------------------------------------- in this macro invocation
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:88:10
>    |
> 88 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "h1-client")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
> 
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:108:9
>     |
>  88 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "h1-client")]
>     |          -------------------------------------------------------------------------------------- in this macro invocation
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:90:10
>    |
> 90 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "h1-client-rustls")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
> 
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:108:9
>     |
>  90 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "h1-client-rustls")]
>     |          --------------------------------------------------------------------------------------------- in this macro invocation
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: cannot find macro `cargo_toml` in the current scope when looking from the crate root
>   --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:92:10
>    |
> 92 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "wasm-client")]
>    |          ^^^^^^^^^^ not found from the crate root
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>    = help: import `macro_rules` with `use` to make it callable above its definition
> 
> warning: cannot find macro `cargo_toml_private` in the current scope when looking from the crate root
>    --> /home/vtvz/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/influxdb-0.7.2/src/lib.rs:108:9
>     |
>  92 | #![doc = cargo_toml!(indent="\t", default-features = false, "derive", "use-serde", "wasm-client")]
>     |          ---------------------------------------------------------------------------------------- in this macro invocation
> ...
> 108 |         cargo_toml_private!($indent, "default-features = false,", $firstfeat $(, $feature)*)
>     |         ^^^^^^^^^^^^^^^^^^ not found from the crate root
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #124535 <https://github.com/rust-lang/rust/issues/124535>
>     = help: import `macro_rules` with `use` to make it callable above its definition
>     = note: this warning originates in the macro `cargo_toml` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
```

</details>

I didn't have this warning in 0.6 version
